### PR TITLE
Enable selecting devices in Rpr

### PR DESCRIPTION
**Pull request by AlexanderVeselov**

### CODE REVIEW

File: Rpr/Utils/config_manager.cpp [hunk 1]
```diff
@@ -58,40 +55,48 @@ void ConfigManager::CreateConfigs(
 
     configs.clear();
 
-    if (req_platform_index >= (int)platforms.size())
-        throw std::runtime_error("There is no such platform index");
-    else if ((req_platform_index > 0) &&
-        (req_device_index >= (int)platforms[req_platform_index].GetDeviceCount()))
-        throw std::runtime_error("There is no such device index");
-
+    std::list<int> req_gpu_indices;
```
Comment (Kvalme) [inline Rpr/Utils/config_manager.cpp]: need to discuss offline

File: Rpr/Utils/config_manager.cpp [hunk 2]
```diff
@@ -58,40 +55,48 @@ void ConfigManager::CreateConfigs(
...
+                    req_gpu_indices.pop_front();
+                }
+                else
+                {
                     continue;
+                }
             }
 
+            if (!use_cpu && (device_type == CL_DEVICE_TYPE_CPU))
+                continue;
```
Comment (Kvalme) [inline Rpr/Utils/config_manager.cpp]: add {}

File: RprTest/basic.h [hunk 3]
```diff
@@ -72,12 +72,47 @@ class BasicTest : public ::testing::Test
         m_reference_path.append("/");
         m_output_path.append("/");
 
-        ASSERT_EQ(rprCreateContext(RPR_API_VERSION, nullptr, 0, RPR_CREATION_FLAGS_ENABLE_GPU0, nullptr, nullptr, &m_context), RPR_SUCCESS);
+        rpr_creation_flags flags = RPR_CREATION_FLAGS_ENABLE_GPU0 | RPR_CREATION_FLAGS_ENABLE_GPU1;//GetCreationFlags();
```
Comment (Kvalme) [inline RprTest/basic.h]: Basic tests should run on single gpu.
Need to add separate test for multi-gpu and for non first gpu

**Review by Kvalme — CHANGES_REQUESTED** [review]

File: RprTest/basic.h (hunk 3 above)
Comment (AlexanderVeselov) [reply RprTest/basic.h]: Ooops, I've forgotten to remove this

**Pull request closed without merging**